converters/resourcegroup_xml_to_yaml.py

```diff
@@ -80,13 +89,29 @@ def add_rg(self, rg: Dict):
         del rg_copy["GroupName"]
 
         try:
-            self.data[sanfacility][sansite][sanrg] = self.simplify_resourcegroup(rg_copy)
+            self.data[sanfacility][sansite][sanrg_filename] = self.simplify_resourcegroup(rg_copy)
         except Exception:
-            print("*** We were parsing %s/%s/%s" % (sanfacility, sansite, sanrg), file=sys.stderr)
+            print("*** We were parsing %s/%s/%s" % (sanfacility, sansite, sanrg_filename), file=sys.stderr)
             pprint.pprint(rg_copy, stream=sys.stderr)
             print("\n\n", file=sys.stderr)
             raise
 
+    def add_downtime(self, downtime: Dict):
+        dt_copy = dict(downtime)
+        del dt_copy["ResourceGroup"]
+        del dt_copy["ResourceFQDN"]  # we can reconstruct this from the ResourceGroup and the ResourceName
+        del dt_copy["ResourceID"]  # ditto
+        del dt_copy["CreatedTime"]
+        del dt_copy["UpdateTime"]
+        dt_copy["Services"] = self.simplify_downtime_services(downtime["Services"])
+        # leading and trailing whitespace causes "\n"'s in the resulting string
+        dt_copy["Description"] = re.sub(r"(?m)[ \t]+$", "", dt_copy["Description"])
+        dt_copy["Description"] = re.sub(r"(?m)^[ \t]+", "", dt_copy["Description"])
+        rgname = downtime["ResourceGroup"]["GroupName"]
+        if rgname not in self.downtimes:
+            self.downtimes[rgname] = []
+        self.downtimes[rgname].append(dt_copy)
```

Comment:
I'm a little confused by this, aren't downtimes per-resource not per-resourcegroup?

Reply:
Yeah but each resource is in a resourcegroup. So each resourcegroup.yaml can have a resourcegroup_downtime.yaml.